### PR TITLE
TINY-8417: Use aria-controls instead of aria-owns in DropdownUtils

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-8417

Description of Changes:

This just cleans up the changes done in https://github.com/tinymce/tinymce/pull/7301/ and adds a couple minor things I found when testing locally.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved
* [x] Rebase merge to keep commit history

GitHub issues (if applicable):
Fixes https://github.com/tinymce/tinymce/issues/7300

### CODE REVIEW

**Review by ltrouton — APPROVED** [review]

File: modules/alloy/src/main/ts/ephox/alloy/alien/ComponentStructure.ts [hunk 1]
```diff
@@ -2,11 +2,14 @@ import { Fun } from '@ephox/katamari';
 import { Compare, PredicateExists, SugarElement } from '@ephox/sugar';
 
 import { AlloyComponent } from '../api/component/ComponentApi';
+import * as AriaControls from '../aria/AriaControls';
 import * as AriaOwner from '../aria/AriaOwner';
 import { AnchorSpec } from '../positioning/mode/Anchoring';
 
 const isAriaPartOf = (component: AlloyComponent, queryElem: SugarElement<Node>): boolean =>
-  AriaOwner.find(queryElem).exists((owner) => isPartOf(component, owner));
+  AriaOwner.find(queryElem)
```
Comment (ephox-mogran) [inline modules/alloy/src/main/ts/ephox/alloy/alien/ComponentStructure.ts:10]: What drove the decision to use either one here? Are we using aria-owner somewhere else? Or just a precaution?

File: modules/alloy/src/main/ts/ephox/alloy/aria/AriaOwner.ts [hunk 2]
```diff
@@ -1,46 +1,7 @@
...
-    Attribute.remove(elem, 'aria-owns');
-  };
-
-  return {
-    id: ariaId,
-    link,
-    unlink
-  };
-};
+const find = AriaManager.find(AriaManager.LinkableAttribute.AriaOwns);
```
Comment (ephox-mogran) [inline modules/alloy/src/main/ts/ephox/alloy/aria/AriaOwner.ts:3]: Minor. I wonder if we should just combine AriaOwner and AriaControls now. I guess it's unclear what to call the combination.

**Review by ephox-mogran — APPROVED** [review]
Ignore my comments. I don't care much about any of them. Nothing to change.

File: modules/alloy/src/main/ts/ephox/alloy/alien/ComponentStructure.ts (hunk 1 above)
Comment (lnewson) [reply modules/alloy/src/main/ts/ephox/alloy/alien/ComponentStructure.ts:10]: Good question... I didn't look into that. I'll see where else it's used, as it's possible we don't need the `AriaOwner` module at all anymore.

Comment (lnewson) [reply modules/alloy/src/main/ts/ephox/alloy/alien/ComponentStructure.ts:10]: So looks like it's used by the `FloatingToolbarButton` schema, for which `aria-owner` would be incorrect there again so we should swap that to use `aria-controls` as well. I'll swap it out and just drop all this back to a single `AriaControls` module.

File: modules/alloy/src/main/ts/ephox/alloy/aria/AriaOwner.ts (hunk 2 above)
Comment (lnewson) [reply modules/alloy/src/main/ts/ephox/alloy/aria/AriaOwner.ts]: It's all the one thing now anyways, as we didn't need the `AriaOwner` module in the end (see my other comment).

**Pull request merged**